### PR TITLE
Implement CICD via GithubActions

### LINKED ISSUE #56: Implement cicd for frontend's unit and integration tests

### PR DESCRIPTION
**Pull request by isabellachen3988**
- Configuration file (UnitAndIntegrationTest.yml) informs GithubActions to run through all tests under androidTest and test
-  This should happen whenever a user either pushes to main & develop, or makes a pull request to main & develop

To test, you can branch off from this branch, add your branch name under `push: branches: [ main, Develop ]` and make a commit/push to make some tests to fail/succeed. The github action should immediately run on your branch after push. 

Closes #56

### CODE REVIEW

**Comment by isabellachen3988** [comment]
Waiting on @KevinZxy1273 to resolve errors on SettingScreenInstrumentedTest

File: app/build.gradle.kts
Comment (KevinZxy1273) [inline app/build.gradle.kts:1]: Added dependency for implementing navigation testing.

File: app/src/androidTest/java/com/hydrofish/app/SettingsScreenInstrumentedTest.kt
Comment (KevinZxy1273) [inline app/src/androidTest/java/com/hydrofish/app/SettingsScreenInstrumentedTest.kt:1]: Added navController and mockUserSessionRepository for upcoming testng.

**Comment by KevinZxy1273** [comment]
Refactored the parameter from a class to an interface to allow for Mockito-based mocking in tests.

**Review by marklhj-lee — APPROVED** [review]
Check what's app

**Review by KevinZxy1273 — APPROVED** [review]
Looks good

**Review by greatfighter — APPROVED** [review]

**Pull request merged**